File: Top 100 Codes/07-Codes related to Matrix/02-Search an element in a matrix/Searchanelementinamatrix.cpp
#include <iostream>
#include <vector>
using namespace std;

bool searchMatrix(vector<vector<int>>& matrix, int target) {
    int n = matrix.size();
    int m = matrix[0].size();

    int row = 0, col = m - 1; // Start from top-right corner

    while (row < n && col >= 0) {
        if (matrix[row][col] == target)
            return true;
        else if (matrix[row][col] > target)
            col--; // Move left
        else
            row++; // Move down
    }

    return false;
}

int main() {
    vector<vector<int>> matrix = {
        {1, 4, 7, 11},
        {2, 5, 8, 12},
        {3, 6, 9, 16},
        {10,13,14,17}
    };

    int target = 5;

    if (searchMatrix(matrix, target))
        cout << "Element found!" << endl;
    else
        cout << "Element not found." << endl;

    return 0;
}

File: Top 100 Codes/07-Codes related to Matrix/02-Search an element in a matrix/Searchanelementinamatrix.py
def searchMatrix(matrix, target):
    n = len(matrix)
    m = len(matrix[0])

    row, col = 0, m - 1  # Start from top-right

    while row < n and col >= 0:
        if matrix[row][col] == target:
            return True
        elif matrix[row][col] > target:
            col -= 1  # Move left
        else:
            row += 1  # Move down

    return False

# Example usage
matrix = [
    [1, 4, 7, 11],
    [2, 5, 8, 12],
    [3, 6, 9, 16],
    [10,13,14,17]
]

target = 5

if searchMatrix(matrix, target):
    print("Element found!")
else:
    print("Element not found.")
